public-packages/llm-client/src/types/index.ts

```diff
@@ -49,19 +56,107 @@ export type AnthropicChatCompletionParams =
   | AnthropicChatCompletionParamsStream
   | AnthropicChatCompletionParamsNonStream
 
-export type OpenAILikeClient<P> = P extends "openai"
+/** Google types */
+export type GoogleChatCompletionParamsStream = Omit<
+  Partial<OpenAI.ChatCompletionCreateParams>,
+  "messages" | "model"
+> & {
+  messages: SupportedChatCompletionMessageParam[]
+  stream: true
+  max_tokens: number
+  additionalProperties?: {
+    cacheName?: string
+  }
+  model: GeminiGenerativeModels | string
+}
+
+export type GoogleChatCompletionParamsNonStream = Omit<
+  Partial<OpenAI.ChatCompletionCreateParams>,
+  "messages" | "model"
+> & {
+  messages: SupportedChatCompletionMessageParam[]
+  stream?: false | undefined
+  max_tokens: number
+  additionalProperties?: {
+    cacheName?: string
+  }
+  model: GeminiGenerativeModels
+}
+
+export type GoogleChatCompletionParams =
+  | GoogleChatCompletionParamsStream
+  | GoogleChatCompletionParamsNonStream
+
+export type ExtendedCompletionGoogle = Partial<OpenAI.ChatCompletion> & {
+  originResponse: EnhancedGenerateContentResponse
+}
+
+export type ExtendedCompletionChunkGoogle = Partial<OpenAI.ChatCompletionChunk> & {
+  originResponse: EnhancedGenerateContentResponse
+}
+
+export type GoogleCacheCreateParams = GoogleChatCompletionParams & {
+  ttlSeconds: number
+}
+
+export type GeminiGenerativeModels =
+  | "gemini-1.5-pro"
+  | "gemini-1.5-pro-latest"
+  | "gemini-1.5-flash-8b"
+  | "gemini-1.5-flash-8b-latest"
+  | "gemini-1.5-flash"
+  | "gemini-1.5-flash-latest"
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  | (string & {})
```

Comment:
Gemini doesn't give you a typed way in their library to see what models are available but this at least gives a good starting point while allowing extensions